packages/dev/core/src/Maths/ThinMaths/thinMath.matrix.functions.ts

```diff
@@ -0,0 +1,267 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import type { IMatrixLike } from "../math.like";
+import type { ThinMatrix } from "./thinMath.matrix";
+
+function SetMatrixData(
+    result: ThinMatrix,
+    m0: number,
+    m1: number,
+    m2: number,
+    m3: number,
+    m4: number,
+    m5: number,
+    m6: number,
+    m7: number,
+    m8: number,
+    m9: number,
+    m10: number,
+    m11: number,
+    m12: number,
+    m13: number,
+    m14: number,
+    m15: number
+): void {
+    const mat = result.asArray();
+    mat[0] = m0;
+    mat[1] = m1;
+    mat[2] = m2;
+    mat[3] = m3;
+    mat[4] = m4;
+    mat[5] = m5;
+    mat[6] = m6;
+    mat[7] = m7;
+    mat[8] = m8;
+    mat[9] = m9;
+    mat[10] = m10;
+    mat[11] = m11;
+    mat[12] = m12;
+    mat[13] = m13;
+    mat[14] = m14;
+    mat[15] = m15;
+
+    result.updateFlag++;

```

Comment:
Currently, we can check that two `Matrix` instances are identical by testing their `updateFlag`: if there are identical, then the matrices are identical, even if the two instances are different (copying a matrix into another will copy the source `updateFlag` into the destination, for eg). It works because we update `Matrix.updateFlag` with an ever increasing static counter whenever we must update this property. It's not the case with `ThinMatrix`, meaning we don't want to support this feature?

Reply:
yes but I have to have it as this is part of the IMatrixLike

Reply:
But we do not use it in the Thin maths